File: TP/back/TpProductService/src/main/java/cda/tpproductservice/entity/Product.java
package cda.tpproductservice.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "product")
public class Product
{
	@Column(name = "name", nullable = false)
	String name;
	@Column(name = "description", nullable = false)
	String description;
	@Column(name = "price", nullable = false)
	double price;
	@Column(name = "stock", nullable = false)
	int stock;
	@Column(name = "category", nullable = false)
	@Enumerated(EnumType.STRING)
	ProductCategory category;
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id", nullable = false)
	private Long id;

	public enum ProductCategory
	{
		CATEGORY_1,
		CATEGORY_2,
		CATEGORY_3
	}
}

File: TP/mobile/src/classes/Product.ts
export class Product {
    id: number;
    category: string;
    description: string;
    name: string;
    price: number;
    imageUrl: string;

    constructor(id: number, category: string, description: string, name: string, price: number, imageUrl: string) {
        this.id = id;
        this.category = category;
        this.description = description;
        this.name = name;
        this.price = price;
        this.imageUrl = imageUrl;
    }
}
